pynequa/models.py

```diff
@@ -1,13 +1,48 @@
 from typing import Dict, List, Optional
-
-
-class TreeParams:
+from abc import abstractmethod
+from dataclasses import dataclass, field
+import logging
+
+logging.basicConfig(level=logging.DEBUG)
```

Comment:
Let's use [loguru](https://github.com/Delgan/loguru) for logging. In-built logging module is pretty obfuscated and difficult to navigate.

Reply:
I wanted to use in-built module as much as possible just to keep this package at its bare minimum. But I can change it to use loguru.